applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_matrix_free.h

```diff
@@ -142,42 +142,51 @@ class MapperVertexMorphingMatrixFree : public Mapper
         mValuesDestination[2].clear();
 
         // Perform mapping
-        for(auto& node_i : mrDestinationModelPart.Nodes())
+        #pragma omp parallel for
+        for(int node_itr=0; node_itr < static_cast<int>(mrDestinationModelPart.NumberOfNodes()); node_itr++)
         {
+            auto node_i = mrDestinationModelPart.NodesBegin() + node_itr;
```

Comment:
Doesn't it work to get `node_i` a reference?

Reply:
Node I here is an iterator. So there is no overhead and you simply use its pointer. Why would you go for a reference here?

Reply:
Three functions that are using it below need a reference. Also the existing code was using references and it just reduces the changes in this PR.